### PR TITLE
Typescript/compatibility

### PR DESCRIPTION
**Pull request by dpinol**
- refactor of router to have smaller methods
- compile core & react with babel
- types for typescript for some core & react symbols
- eslint & prettier config

### CODE REVIEW

File: packages/botonic-core/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface Route {
+  path: string;
+  payload?: RegExp| string;
+  text?: string;
+  action: React.ReactNode;
+}
```
Comment (guinii) [inline packages/botonic-core/index.d.ts]: If this component is representing the 'router' of Botonic, we need ti add intent variable, I don't remember right now but we had the option of passing an attribute called 'type', where we defined the type of the input, like: 'type: image', or 'type: text'. I don't know if it's necessary to declare it, but to keep in mind.

File: packages/botonic-react/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+export const RequestContext: React.Context<any>;
+
+export interface Session {
+  __locale: string;
+  __retries?: number;
+}
+
+// plugins
+export interface PluginPreInput {
```
Comment (guinii) [inline packages/botonic-react/index.d.ts]: The plugin, in the pre input, why can't return a response? As it's done in the PluginPostInput?

Comment (dpinol) [reply packages/botonic-react/index.d.ts]: it's actually the type for the pre() method. I added comment in future PR

File: packages/botonic-core/index.d.ts (hunk 1 above)
Comment (dpinol) [reply packages/botonic-core/index.d.ts]: right! thanks
I'm adding this in future PR
```
type StringMatcher = RegExp| string | ((data: string) => boolean);

export interface Route {
  path: StringMatcher;
  payload?: StringMatcher;
  text?: StringMatcher;
  type?: StringMatcher;
  intent?: StringMatcher;
  session?: (session: any) => boolean;
  action: React.ReactNode;
}
```

**Pull request merged**